Support/Checkpoints/CheckpointReader.swift

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import TensorFlow
+
+/// A shell for what will be a Swift-native checkpoint reader. Currently supplements the internal 
+/// C API TensorFlow checkpoint reader with the ability to download remote checkpoints.
+open class CheckpointReader {
+    fileprivate let reader: TensorFlowCheckpointReader
+
+    /// The local checkpoint location.
+    public let localCheckpointLocation: URL
+
+    /// The number of tensors stored in the checkpoint.
+    public var tensorCount: Int { reader.tensorCount }
+
+    /// The names of the tensors stored in the checkpoint.
+    public var tensorNames: [String] { reader.tensorNames }
+
+    /// Initializes the checkpoint reader from either a local or remote directory. If remote, 
+    /// automatically downloads the checkpoint files into a temporary directory.
+    ///
+    /// - Arguments:
```

Comment:
Minor: `Parameters` is the proper API doc comment that's understood by doc generation tools. 
```suggestion
    /// - Parameters:
```